pandas/tests/frame/test_operators.py

```diff
@@ -671,6 +671,28 @@ def _test_seq(df, idx_ser, col_ser):
         exp = DataFrame({'col': [False, True, False]})
         assert_frame_equal(result, exp)
 
+    def test_return_dtypes_bool_op_costant(self):
+        # GH15077
+        df = DataFrame({'x': [1, 2, 3], 'y': [1., 2., 3.]})
+        const = 2
+
+        # not empty DataFrame
+        for op in ['eq', 'ne', 'gt', 'lt', 'ge', 'le']:
+            f = getattr(df, op)
+            for col in ['x', 'y']:
+                res = f(const)
+                c = getattr(res, col)
+                self.assertEqual(c.dtypes, bool)
+
+        # empty DataFrame
+        empty = df.iloc[:0]
+        for op in ['eq', 'ne', 'gt', 'lt', 'ge', 'le']:
+            f = getattr(empty, op)
+            for col in ['x', 'y']:
```

Comment:
instead of this comparison, use ``assert_series_equal`` and compare against ``Series([2], ['bool']``
where you do ``result = getattr(empty, op)(const).get_dtype_counts()``
e.g
```
In [5]: df = DataFrame({'x': [1, 2, 3], 'y': [1., 2., 3.]})

In [6]: df.lt(2)
Out[6]: 
       x      y
0   True   True
1  False  False
2  False  False

In [7]: df.lt(2).dtypes
Out[7]: 
x    bool
y    bool
dtype: object

In [8]: df.lt(2).dtypes.values
Out[8]: array([dtype('bool'), dtype('bool')], dtype=object)

In [9]: df.lt(2).get_dtype_counts()
Out[9]: 
bool    2
dtype: int64

In [10]: Series([2], ['bool'])
Out[10]: 
bool    2
dtype: int64
```